packages/core/src/components/dropdown/dropdown.tsx

```diff
@@ -83,27 +83,67 @@ export class TdsDropdown {
     this.handleChange();
   }
 
-  /** Method for setting the value of the Dropdown. */
+  /** Method for setting the value of the Dropdown.
+   *
+   * Single selection example:
+   *
+   * <code>
+   *  dropdown.setValue('option-1', 'Option 1');
+   * </code>
+   *
+   * Multiselect example:
+   *
+   * <code>
+   *  dropdown.setValue([<br />
+   *  &nbsp;{ value: 'option-4', label: 'Option 4' },<br />
+   *  &nbsp;{ value: 'option-1', label: 'Option 1' }<br />
+   *  ]);
+   * </code>
+   */
   @Method()
-  async setValue(newValue: string, newValueLabel: string) {
-    if (this.multiselect) {
-      this.selection = this.selection
-        ? [...this.selection, { value: newValue, label: newValueLabel }]
-        : [{ value: newValue, label: newValueLabel }];
-    } else {
-      this.selection = [{ value: newValue, label: newValueLabel }];
-      this.children = this.getChildren().map((element: HTMLTdsDropdownOptionElement) => {
-        if (element.value !== newValue) {
-          element.setSelected(false);
-        }
-        return element;
-      });
+  async setValue(
+    value: string | { value: string; label: string } | { value: string; label: string }[],
+    label?: string,
+  ) {
+    let nextValue: Array<{ value: string; label: string }>;
+    if (typeof value === 'string') nextValue = [{ value, label }];
+    else if (!Array.isArray(value)) nextValue = [value];
+    else nextValue = value;
+
+    if (!this.multiselect && nextValue.length > 1) {
+      console.warn('Tried to select multiple items, but multiselect is not enabled.');
+      nextValue = [nextValue[0]];
     }
-    this.handleChange();
+
+    this.reset();
```

Comment:
Since this will emit a `tdsChange` event, should we do an "inline"-reset here istead? So reset without calling the function? @alexandereneroth

Reply:
Yea, I started on somehting like this, what do you think?
I think it makes sense to reuse code like this, but naming could perhaps improve.

```

  private internalReset() {
    this.children = this.getChildren().map((element: HTMLTdsDropdownOptionElement) => {
      element.setSelected(false);
      return element;
    });
    this.selection = null;
    this.host.setAttribute('value', null);
  }

  /** Method that resets the Dropdown, marks all children as non-selected and resets the value to null. */
  @Method()
  async reset() {
    this.internalReset()
    this.handleChange();
  }
  ```

Reply:
(and then use internalReset() instead of reset())

Reply:
Love it! Ping me when you've committed it 🚀